lib/chai/utils/addChainableMethod.js

```diff
@@ -82,13 +82,24 @@ module.exports = function addChainableMethod(ctx, name, method, chainingBehavior
         chainableBehavior.chainingBehavior.call(this);
 
         var chainableMethodWrapper = function () {
-          // Use this chainable method wrapper as the starting point for
-          // removing implementation frames from the stack trace of a failed
-          // assertion. Note that this is the correct starting point even if
-          // this assertion has been overwritten since overwriting a chainable
-          // method merely replaces the saved methods in `ctx.__methods` instead
-          // of completely replacing the overwritten assertion.
-          flag(this, 'ssfi', chainableMethodWrapper);
+          // Setting the `ssfi` flag to `chainableMethodWrapper` causes this
+          // function to be the starting point for removing implementation
+          // frames from the stack trace of a failed assertion.
+          //
```

Comment:
The `ssfi` flag should always contain a reference to the lowest internal Chai frame in the stack, because that frame and all frames above it will be filtered out. In other words, the very first Chai internal function that gets hit in an assertion should set the `ssfi` flag, and any functions after that shouldn't mess with the `ssfi` flag.

For example:

```js
expect({foo: 1}).to.include({foo: 1});
```

- When `expect({foo: 1})` gets called, `ssfi` is `undefined`.
- When `.to` gets called, `ssfi` gets set to either the `proxyGetter` or `propertyGetter` function for `.to`, depending on whether on not proxies are supported and enabled. Note that `.to` doesn't call any more internal Chai functions, so we're done with this step. But if it did happen to call more internal Chai functions, then it'd freeze it's `ssfi` before calling them since we want to keep `ssfi` pointing at the first internal Chai function that gets hit when `.to` is invoked. 
- When `.include` gets called, `ssfi` gets set to the `chainableMethodWrapper` function for `.include`. Now, `.include` calls the `.property` assertion internally, so we freeze `ssfi` before calling `.property` since we want to keep `ssfi` pointing at the first internal Chai function that gets hit when `.include` is invoked.